src/memote/suite/tests/test_consistency.py

```diff
@@ -51,32 +51,82 @@ def test_stoichiometric_consistency(model):
     publication. In addition, the list of min unconservable sets is computed
     using the algorithm described in section 3.3.
 
+    """
+    ann = test_stoichiometric_consistency.annotation
+    is_consistent = consistency.check_stoichiometric_consistency(model)
+    ann["data"] = is_consistent
+    ann["metric"] = 1.0 - float(is_consistent)
+    ann["message"] = wrapper.fill(
+        """This model is stoichiomerically {}""".format(
+            "consistent" if is_consistent else "inconsistent"
+        )
+    )
+    assert is_consistent, ann["message"]
+
+
+@annotate(title="Uncoserved Metabolites", format_type="percent")
+def test_unconserved_metabolites(model):
+    """
+    Report number of unconserved metabolites (if model incosistent).
+
+    List of unconserved metabolites is computed using the algorithm described
+    in section 3.2 of
+    "Detection of Stoichiometric Inconsistencies in Biomolecular Models."
+    Bioinformatics 24, no. 19 (2008): 2245.
+    doi: 10.1093/bioinformatics/btn425.
     """
     ann = test_stoichiometric_consistency.annotation
     is_consistent = consistency.check_stoichiometric_consistency(model)
     ann["data"] = {
         "unconserved_metabolites": []
         if is_consistent
         else get_ids(consistency.find_unconserved_metabolites(model)),
+    }
+    ann["metric"] = len(ann["data"]["unconserved_metabolites"]) / len(model.metabolites)
+    ann["message"] = wrapper.fill(
+        """This model contains {} ({:.2%}) unconserved
+        metabolites: {}""".format(
+            len(ann["data"]["unconserved_metabolites"]),
+            ann["metric"],
+            truncate(ann["data"]["unconserved_metabolites"]),
+        )
+    )
+    # equivalent to is_consistent
+    assert ann["metric"] == 0, ann["message"]
+
+
+@annotate(title="Minimal Inconsistent Stoichiometries", format_type="count")
+def test_inconsistent_min_stoichiometry(model):
```

Comment:
I just looked into the paper again and I think this test on its own actually doesn't deliver any insight. Only taken together with the elementary leakage modes described in section 3.4 do we find out more, i.e., the actual set of reactions contributing to the inconsistency.

> As an alternative solution for large models, we propose, instead of
calculating a complete set of elementary modes, calculating only a spanning
set that comprises the columns of a nullspace matrix of the system (N|y)v = 0;
this matrix can be obtained by Gauss–Jordan elimination. The inspection of
this matrix enables the detection of at least some, if not all of the input
errors causing the leakage of a given metabolite set (Fig. 2d). After the error
correction, the matrix can be recalculated to detect the remaining errors.

Would you be up for adding that? Probably best handled in a separate PR so that this can proceed.

Reply:
Maybe it is not that informative for the user without the modes. I can try to do that in another PR.

Reply:
Great, that'd be stellar!